### ISSUE TITLE
[Regression] pilot upload fails in 2.69.0: private method 'include' called for Array:Class (NoMethodError)

### ISSUE THREAD

**Issue by MikePaer** [fastlane/fastlane#11199]
### New Issue Checklist

- [✔] Updated fastlane to the latest version
- [✔] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [✔] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [✔] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
I submit app builds to TestFlight through Visual Studio Team Services continuous integration, which uses Fastlane. The CI system is set up to automatically update Fastlane if an update is available for each release. That's how I found this regression.

Yesterday, the release worked fine. Checking the logs, I found that testflight version `2.68.2` was used then. Today's release failed (it had updated TestFlight to `2.69.0`).  It reliably fails each time on this new version.

Note: in the meanwhile, I have configured the CI system to force Fastlane version `2.68.2` for now to work around this issue.

Here's the command the CI system runs, and the failing output, when using Fastlane `2.69.0`.

##### Complete output when running fastlane, including the stack trace and command used
`fastlane pilot upload -u ******** -i /Users/*******/buildagent/_work/r5/a/*******/ios_release/******.ipa -a com.*****.******* --skip_waiting_for_build_processing true`

```
fastlane pilot upload -u ******** -i /Users/*******/buildagent/_work/r5/a/*******/ios_release/******.ipa -a com.*****.******* --skip_waiting_for_build_processing true
2017-12-12T23:42:13.4643610Z /Users/xamarin/.gem-cache/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core/core_ext/cfpropertylist.rb:17:in `block in <top (required)>': private method `include' called for Array:Class (NoMethodError)
2017-12-12T23:42:13.4743770Z 	from /Users/xamarin/.gem-cache/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core/core_ext/cfpropertylist.rb:14:in `each'
2017-12-12T23:42:13.4846830Z 	from /Users/xamarin/.gem-cache/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core/core_ext/cfpropertylist.rb:14:in `<top (required)>'
2017-12-12T23:42:13.4945890Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-12-12T23:42:13.5052070Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-12-12T23:42:13.5149200Z 	from /Users/xamarin/.gem-cache/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core/ipa_file_analyser.rb:2:in `<top (required)>'
2017-12-12T23:42:13.5179650Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-12-12T23:42:13.5209300Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-12-12T23:42:13.5236760Z 	from /Users/xamarin/.gem-cache/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core.rb:19:in `<top (required)>'
2017-12-12T23:42:13.5266310Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-12-12T23:42:13.5295180Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-12-12T23:42:13.5325740Z 	from /Users/xamarin/.gem-cache/gems/fastlane-2.69.0/fastlane/lib/fastlane.rb:1:in `<top (required)>'
2017-12-12T23:42:13.5357100Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-12-12T23:42:13.5396980Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-12-12T23:42:13.5427300Z 	from /Users/xamarin/.gem-cache/gems/fastlane-2.69.0/fastlane/lib/fastlane/cli_tools_distributor.rb:18:in `take_off'
2017-12-12T23:42:13.5457120Z 	from /Users/xamarin/.gem-cache/gems/fastlane-2.69.0/bin/fastlane:20:in `<top (required)>'
2017-12-12T23:42:13.5484290Z 	from /usr/local/bin/fastlane:23:in `load'
2017-12-12T23:42:13.5510660Z 	from /usr/local/bin/fastlane:23:in `<main>'
2017-12-12T23:42:13.5714150Z ##[error]Error: fastlane failed with return code: 1
```

### Environment

(I can't easily run this command since Fastlane is run through a continuous integration system. Let me know if you need this and I'll figure out a way.)

**Comment by cbo1964** [comment]
I have the same issue. Yesterday was working, today I updated to last version and now it is not working giving the following error:
/Library/Ruby/Gems/2.0.0/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core/core_ext/cfpropertylist.rb:17:in `block in <top (required)>': private method `include' called for Array:Class (NoMethodError)
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core/core_ext/cfpropertylist.rb:14:in `each'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core/core_ext/cfpropertylist.rb:14:in `<top (required)>'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core/ipa_file_analyser.rb:2:in `<top (required)>'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core.rb:19:in `<top (required)>'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.69.0/fastlane/lib/fastlane.rb:1:in `<top (required)>'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.69.0/fastlane/lib/fastlane/cli_tools_distributor.rb:18:in `take_off'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.69.0/bin/fastlane:20:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'

**Comment by milch** [comment]
I just started investigating this, sorry about the issues! It seems to come from an older version of ruby (that we still support). I double checked the documentation for ruby and the method *should* exist, so this was likely a bug that got fixed in a later version. I'm exploring alternative options for the code that's failing now

**Comment by milch** [comment]
@MikePaer @cbo1964 What system versions are you using? Am I correct to assume that both of you are using the system ruby?

**Comment by trusk89** [comment]
I can confirm the issue. Started showing up like 10 hours ago on our jenkins machine. Probably someone updated fastlane.

`workspace/gems/ruby/2.0.0/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core/core_ext/cfpropertylist.rb:17:in `block in <top (required)>': private method `include' called for Array:Class (NoMethodError)
	from /Users/jenkins/.jenkins/jobs/DEP/jobs/iOS_PullRequester_RD/workspace/gems/ruby/2.0.0/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core/core_ext/cfpropertylist.rb:14:in `each'
	from /Users/jenkins/.jenkins/jobs/DEP/jobs/iOS_PullRequester_RD/workspace/gems/ruby/2.0.0/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core/core_ext/cfpropertylist.rb:14:in `<top (required)>'
	from /Users/jenkins/.jenkins/jobs/DEP/jobs/iOS_PullRequester_RD/workspace/gems/ruby/2.0.0/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core/ipa_file_analyser.rb:2:in `require'
	from /Users/jenkins/.jenkins/jobs/DEP/jobs/iOS_PullRequester_RD/workspace/gems/ruby/2.0.0/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core/ipa_file_analyser.rb:2:in `<top (required)>'
	from /Users/jenkins/.jenkins/jobs/DEP/jobs/iOS_PullRequester_RD/workspace/gems/ruby/2.0.0/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core.rb:19:in `require'
	from /Users/jenkins/.jenkins/jobs/DEP/jobs/iOS_PullRequester_RD/workspace/gems/ruby/2.0.0/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core.rb:19:in `<top (required)>'
	from /Users/jenkins/.jenkins/jobs/DEP/jobs/iOS_PullRequester_RD/workspace/gems/ruby/2.0.0/gems/fastlane-2.69.0/fastlane/lib/fastlane.rb:1:in `require'
	from /Users/jenkins/.jenkins/jobs/DEP/jobs/iOS_PullRequester_RD/workspace/gems/ruby/2.0.0/gems/fastlane-2.69.0/fastlane/lib/fastlane.rb:1:in `<top (required)>'
	from /Users/jenkins/.jenkins/jobs/DEP/jobs/iOS_PullRequester_RD/workspace/gems/ruby/2.0.0/gems/trainer-0.7.0/lib/trainer.rb:1:in `require'
	from /Users/jenkins/.jenkins/jobs/DEP/jobs/iOS_PullRequester_RD/workspace/gems/ruby/2.0.0/gems/trainer-0.7.0/lib/trainer.rb:1:in `<top (required)>'
	from /Users/jenkins/.jenkins/jobs/DEP/jobs/iOS_PullRequester_RD/workspace/gems/ruby/2.0.0/gems/trainer-0.7.0/bin/trainer:4:in `require'
	from /Users/jenkins/.jenkins/jobs/DEP/jobs/iOS_PullRequester_RD/workspace/gems/ruby/2.0.0/gems/trainer-0.7.0/bin/trainer:4:in `<top (required)>'
	from ./bin/trainer:21:in `load'
	from ./bin/trainer:21:in `<main>'
osxbuildserver:workspace jenkins$ 
`

**Comment by milch** [comment]
I just submitted a PR #11202 that should fix this issue. 

If you don't want to wait until this is released, you can try my branch by using a `Gemfile`. Create a `Gemfile` in your project directory: 

```ruby
gem "fastlane", github: "milch/fastlane", branch: "regression/fix-plist-for-old-rubies"
```

Make sure to install bundler if you don't have it already. Then, in your command line, run `bundle install` and you will be able to run fastlane by using `bundle exec`: 

```bash
bundle exec fastlane frameit
```

To get back to a regular release version of fastlane you can simply remove the `Gemfile`, or remove the `github` and `branch` parameters from the `Gemfile` I presented above to read:

```ruby
gem "fastlane"
```

If you removed the `Gemfile` simply remove the `bundle exec` from your fastlane commands. Otherwise you can re-run `bundle install` after you've made the above edits.

Let me know if you have any questions!

**Comment by trusk89** [comment]
Thanks @milch for the fast fix! For now I've downgraded to 2.68 but it's good to have an alternative until the fix gets merged!

**Comment by cbo1964** [comment]
Hi,
sorry but I'm not expert enough to change my fastlane configuration to test your fix.

By the way, I could I downgrade to previous version ?

**Comment by milch** [comment]
@cbo1964 If you installed fastlane via rubygems, you can do that by running `gem install --version 2.68.0 fastlane`. The new version should be released shortly though

**Comment by cbo1964** [comment]
many thanks
Claudio

**Comment by milch** [comment]
Hey everyone! As you might've seen we've released 2.69.1 a couple hours ago. Let me know if you have any more issues with this.

**Comment by cbo1964** [comment]
Hi,

I just tested it. It works. 

Many thanks

Claudio
